### ISSUE TITLE
Using signals package with react router results in a wierd jsx error

### ISSUE THREAD

**Issue by salamaashoush** [preactjs/signals#251]
I tried using `@preact/signals-react` in a project with react-router 6 but I got that error which doesn't make any sense to me, the error is complaining about react-router but originates from `@preact/signals-react` and only happens when I import/use anything from `@preact/signals-react`

<img width="882" alt="image" src="https://user-images.githubusercontent.com/13505298/197622980-87336b17-49bf-488b-af7b-8f4c8725bcb8.png">

here is a link https://stackblitz.com/edit/react-router-starter-gfxkdm?file=components%2FHome.js  to reproduce the error, try commenting out the signal import/usage in Home.js and everything will work fine 

I think it could be related to the patches to JSX factories, which breaks the react-router check for the `<Route/>`  element type somehow, if I used it with the routing config approach from react-router 6.4 it works fine

**Comment by salamaashoush** [comment]
couldn't find a workaround for this error so I had to use the `@preact/signals-core` package with this custom hook
```tsx
export function useSignalValue<T>(signal: Signal<T>) {
  const [value, setValue] = useState<T>(() => signal.peek());
  useEffect(() => signal.subscribe(setValue), [signal]);
  return value;
}
```

**Comment by vthorsteinsson** [comment]
I have the same error, using `@react-navigation/native` (6.0.13) and `@react-navigation/stacks` (6.3.2). Signals seems to insert itself into the list of children of the Stack.Navigator component, causing it to trigger a runtime exception. Signals thus cannot presently be used in RN apps that rely on `react-navigation`.

**Comment by lennerd** [comment]
Problem is the proxy around every React element type (the actual components) to enable rerending when signals change. So the proxy around the `Route` component breaks the `element.type === Route` check inside `react-router`. @developit Interested in how you plan to fix this. Strict equality check will always fail here, no?

**Comment by developit** [comment]
@lennerd the fix may be to go back to injecting updaters into components via react internals rather than wrapping all components. I've been using and testing a local version of that approach this week.

**Comment by gokul-waterlabs** [comment]
Am able to solve the issue by rolling back the router version to 6.0.0

**Comment by salamaashoush** [comment]
@developit  do you have any plan to publish your local work/fix?

**Comment by lennerd** [comment]
@developit uff, okay. Just had another, quite similar issue, which was really hard to debug (actually was revisting it again after weeks).

Essentially React was warning me about an unmounted component updating its state. Via the debugger I was able to see why: my root component (`HistoryRouter` from `react-router-dom`) was unmounting the complete tree underneath it. Turns out: I was lazy loading `@preact/signals-react`. So when triggering a navigation and rerendering the root component (`HistoryRouter`) React essentially found new child component type (a proxy from `@preact/signals-react`) and unmounted the whole thing.

So long story short: what is your progress with the fix? 😬

**Comment by developit** [comment]
I've been using this "lite" version of the React adapter in the meantime, which doesn't include the Signals optimizations, but works with all React libraries and versions:
https://gist.github.com/developit/f957536487a84dee334cfeb53748f865

@lennerd as a fix for the lazy-loading issue, you can add a static import for `@preact/signals-react` to your root component (`<App />`).

**Comment by lennerd** [comment]
@developit Thats what I'm doing right now and its working just fine. Still not quite what I was hoping for when doing code splitting. 😉 

In any case, thanks for the gist! Nice to see how easy the core package can be included in React. I currently have a quite complex setup with multiple class having signal and computed properties, so it's easier and less error prone to use the original package. And work around the error described here until some fix was released.

**Comment by jibinmathew69** [comment]
@developit Any updates to fix the issue?

**Comment by jibinmathew69** [comment]
As @gokul-waterlabs suggested, I tried using router 6.0.0 and it seems to work, any explainations on why it worked?

**Comment by sunderbharath85** [comment]
Any update on fix?

**Comment by StephenSHorton** [comment]
+1 for fix on this! ❤️

**Comment by developit** [comment]
@jaydenintusurg in the lite version from my gist, you can use `signal()` anywhere, then pass those values through `useSignal()` to access them. You don't have to create signals with the hooks, only access them.
